### PR TITLE
add a unit test sample

### PR DESCRIPTION
**Pull request by shawnkaku**

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
 const express = require('express'); //require express module
 const linebot = require('linebot');
+const app = express(); //creates an Express application
+const http = require('http');
```
Comment (kong1191) [inline App.js]: can be removed due to express framework already done this for you

File: App.js [hunk 2]
```diff
@@ -40,6 +44,9 @@ const port = process.env.PORT || 3000;
 //whatever is in the environment variable 'PORT', or 3000 if there's nothing there
 
 //binds and listens for connections on the specified host and port. This method is identical to Node’s http.Server.listen().
-app.listen(port);
-
-console.log(`MERN Stack listening on ${port}`);
+// app.listen(port);
+// console.log(`MERN Stack listening on ${port}`);
+http.createServer(app)
```
Comment (kong1191) [inline App.js]: use app.listen() instead of creating http server by yourself

File: App.js [hunk 3]
```diff
@@ -25,7 +28,8 @@ bot.on('message', function (event) {
 //Express API --------- App.get('path', callback function);
 //routes HTTP GET requests to the specified path with the specified callback functions
 app.get('/', function (request, response) {
-    response.send("response from node service");
+    // response.send("response from node service");
```
Comment (kong1191) [inline App.js]: do not leave old code as comment

File: App.js [hunk 4]
```diff
@@ -1,14 +1,17 @@
 const express = require('express'); //require express module
 const linebot = require('linebot');
+const app = express(); //creates an Express application
```
Comment (kong1191) [inline App.js]:
```suggestion
var app = module.exports = express();
```

File: App.js [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
+const http = require('http');
+// for test
+global.request = require('supertest');
+request = request(app);
```
Comment (kong1191) [inline App.js]: The above two lines should be moved to the test file (test/App.js)

File: test/App.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+const { expect } = require('chai');
```
Comment (kong1191) [inline test/App.js]: Add the following two lines to import app object:
```
var app = require('../App.js')
var request = require('supertest')
```

File: test/App.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+describe('# This is a test sample.', () => {
+  it('This is a test for /.', (done) => {
+    request.get('/').end((err, res) => {
```
Comment (kong1191) [inline test/App.js]:
```suggestion
    request(app).get('/').end((err, res) => {
```

File: App.js (hunk 4 above)
Comment (shawnkaku) [reply App.js]: will fix it.

File: App.js (hunk 1 above)
Comment (shawnkaku) [reply App.js]: will fix it.

File: App.js (hunk 5 above)
Comment (shawnkaku) [reply App.js]: will fix it.

File: App.js (hunk 3 above)
Comment (shawnkaku) [reply App.js]: will fix it.

File: App.js (hunk 2 above)
Comment (shawnkaku) [reply App.js]: will fix it.

File: test/App.js (hunk 6 above)
Comment (shawnkaku) [reply test/App.js]: will fix it.

File: test/App.js (hunk 7 above)
Comment (shawnkaku) [reply test/App.js]: will fix it.

File: test/mocha.opts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+--timeout 500000
+--require should
+--recursive ./App.js
```
Comment (kong1191) [inline test/mocha.opts]: This line should be removed

File: test/mocha.opts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+--timeout 500000
```
Comment (kong1191) [inline test/mocha.opts]: 5 minutes is enough for finishing all the test cases
```suggestion
--timeout 300s
```

**Review by kong1191 — APPROVED** [review]

**Pull request merged**